packages/server/src/shared/internal/serialize.ts

```diff
@@ -18,18 +18,30 @@ type IsAny<T> = 0 extends T & 1 ? true : false;
 // support it as both a Primitive and a NonJsonPrimitive
 type JsonReturnable = JsonPrimitive | undefined;
 
-// prettier-ignore
+/* prettier-ignore */
 export type Serialize<T> =
   IsAny<T> extends true ? any :
+  unknown extends T ? unknown :
   T extends JsonReturnable ? T :
   T extends Map<any, any> | Set<any> ? object :
   T extends NonJsonPrimitive ? never :
   T extends { toJSON(): infer U } ? U :
   T extends [] ? [] :
   T extends [unknown, ...unknown[]] ? SerializeTuple<T> :
   T extends readonly (infer U)[] ? (U extends NonJsonPrimitive ? null : Serialize<U>)[] :
-  T extends object ? Simplify<SerializeObject<UndefinedToOptional<T>>> :
-  never;
+  T extends object ?
+      // Record<K, V> is a special case where we need to ensure that:
+      // 1. we don't match narrower objects like { foo: string } that are also assignable to Record<string, string>
+      // 2. the keys are not symbols
+      // Excluding literal keys (e.g. "foo") from PropertyKey (string | number | symbol) does not exclude anything...
+      Exclude<PropertyKey, keyof T> extends infer TheKeys extends PropertyKey ?
```

Comment:
Could probably use something like `type IsRecord<T> = Record<never, never> extends T ? true : false` for this to make the check simpler